### ISSUE TITLE
corrupt layer in private docker registry

### ISSUE THREAD

**Issue by silarsis** [docker/docker-registry#929]
Heya,

We have someone here who was running docker version 0.6 (I don't know why) and managed to push a corrupt layer as part of ubuntu:14.10 into our registry.

It took us a while to figure out what had gone wrong, but we're now trying to work out how to clean it up. The docker-registry currently has no way to delete a layer that we can see.

Are there plans for delete end points in the API (old or new)? and/or the ability to force push a layer to overwrite what's there currently?

Finally, is there any planned capability to include checksums on layers so as to hopefully dodge this in the future? Ideally as part of the name, so a referenced layer can't contain broken or malicious data?

Thanks,

**Comment by dmp42** [comment]
TLDR: yes :)

More infos in https://github.com/docker/distribution with the registry go-rewrite, and/or support on irc #docker-distribution
